src/Nancy/Diagnostics/DiagnosticsHook.cs

```diff
@@ -57,7 +57,7 @@ public static void Enable(IPipelines pipelines, IEnumerable<IDiagnosticsProvider
                 new RouteResolverTrie(new TrieNodeFactory(routeSegmentConstraints)),
                 environment);
 
-            var serializer = new DefaultObjectSerializer();
+            var serializer = new DefaultObjectSerializer(new AppDomainAssemblyCatalog());

```

Comment:
Because this will have to be replaced on other platforms, like DNX. There's also a caching concern here.